module-system/sov-cli/src/wallet_state.rs

```diff
@@ -52,6 +53,72 @@ impl<Tx: Serialize + DeserializeOwned, Ctx: sov_modules_api::Context> WalletStat
     }
 }
 
+/// A struct representing private key and associated address
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(bound = "Ctx::Address: Serialize + DeserializeOwned")]
+pub struct PrivateKeyAndAddress<Ctx: sov_modules_api::Context> {
+    /// Private key of the address
+    pub private_key: Ctx::PrivateKey,
+    /// Address associated from the private key
+    pub address: Ctx::Address,
+}
+
+impl<Ctx: sov_modules_api::Context> PartialEq for PrivateKeyAndAddress<Ctx> {
+    fn eq(&self, other: &Self) -> bool {
+        self.private_key.pub_key() == other.private_key.pub_key() && self.address == other.address
```

Comment:
This is not very obvious. Usually, private keys can have different representation, especially if with projective points, and secret keys have canonical representation. Here we go the other way around: we generate a pub key and compare equality.

Is the reason for that a limination on `dalek-ed25519`? If so, can we add a comment here explaining that? Or other comment explaining the reasoning for this

Reply:
I am not sure, that we ever had comparing private keys. To be honest, I've added this trait for testing purposes, so maybe it will be safer just to remove it and do explicit comparison in test!

Reply:
Removed this implementation